### PR TITLE
Fix Dockerfile to work again

### PR DESCRIPTION
**Pull request by gregkare**
This now works on my machine (TM). It took me way too long because I don't know how to Docker

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,14 +1,23 @@
-FROM nodesource/node:trusty
+FROM nodesource/trusty:4.4.7
```
Comment (bkero) [inline Dockerfile]: Do we want to pin to a particular nodesource version? Would 'latest' work here, or 5?

https://hub.docker.com/r/nodesource/trusty/

File: Dockerfile [hunk 2]
```diff
@@ -1,14 +1,23 @@
...
 MAINTAINER Ben Kero <ben.kero@gmail.com>
 
-ADD package.json /tmp/package.json
 RUN apt-get update && apt-get install -y git-core
-RUN cd /tmp && npm install
-RUN mkdir -p /hyperchannel && cp -a /tmp/node_modules /hyperchannel/
+
+# The Docker image now sets NODE_ENV="production". All our dependencies are
+# devDependencies
+ENV NODE_ENV development
```
Comment (bkero) [inline Dockerfile]: I would suggest putting this near the command that this environment variable affects

**Comment by skddc** [comment]
The error message for the failed build is rather useless. But I dont' think changing the Dockerfile can break the Ember test suite.

**Comment by gregkare** [comment]
I didn't see the failed build. I restarted it, it's doing the same thing (nothing happens for 10 minutes, Travis cancels it)

**Comment by gregkare** [comment]
Now that the Sockethub client is downloaded from Sockethub [the hardcoded localhost Sockethub](https://github.com/67P/hyperchannel/blob/6bbe907bcd3413fdcca45c33515d8ae29273e6ab/app/index.html#L21-L23) cannot work inside of Docker (since sockethub is running in another container). I'm going to look at how to make this configurable (per environment maybe?)

**Comment by skddc** [comment]
> Now that the Sockethub client is downloaded from Sockethub the hardcoded localhost Sockethub cannot work inside of Docker (since sockethub is running in another container)

I thought that's what container linking is for. Pretty sure this is not just possible, but a usual scenario with Docker Compose.

**Comment by gregkare** [comment]
@skddc Loading it through localhost doesn't work if you're using Docker Machine (you need to manually redirect ports so it's really not convenient). Loading those JS files from localhost should however work on Linux, can you run `docker-compose up` in the orchestration repo?

I need to set up a Linux VM for "native" Docker support but it's using so much disk space I need to clear some space on my laptop first

**Comment by skddc** [comment]
Any news on this one? Should I still test this somehow? (Have to fix my Docker daemon since moving to a new machine broke it somehow).

**Comment by gregkare** [comment]
We should probably wait until we're not on LTE anymore to test this again

**Comment by skddc** [comment]
I bought enough bandwidth. But that wasn't the question anyway.

**Comment by gregkare** [comment]
I'm going to look at how this works with the current master

**Comment by skddc** [comment]
Whoever just wants to run this locally with a remote Sockethub can use `EMBER_ENV=production npm start` by the way. I'm not sure we need Hyperchannel running within Docker for development, and we deploy it to 5apps Deploy for production.

**Comment by skddc** [comment]
Closing this PR for now, leaeving the branch intact.

**Pull request closed without merging**